domino/api/src/main/java/io/quarkus/domino/ProjectDependencyResolver.java

```diff
@@ -923,14 +928,19 @@ private void addImportedBomsToBuild(ArtifactDependency pomArtDep, Model model, M
     }
 
     private String resolveProperty(String expr, org.apache.maven.model.Dependency dep, Map<String, String> props) {
-        if (expr.startsWith("${") && expr.endsWith("}")) {
-            final String name = expr.substring(2, expr.length() - 1);
-            final String value = props.get(name);
-            if (value == null) {
-                log.warn("Failed to resolve " + value + " from " + dep);
+        if (expr.contains("${") && expr.contains("}")) {
+            Matcher matcher = PROPERTY_PATTERN.matcher(expr);
+            String resolvedValue = null;
+            if (matcher.matches()) {
+                String name = matcher.group(2);
+                String value = props.get(name);
```

Comment:
We should probably check whether the `value` is null?

Reply:
we should :-) thanks! added the check